### DISCUSSION TITLE
Authentication and authorization implementation

### DISCUSSION THREAD

**Discussion by g2giovanni** [stac-utils/stac-fastapi#698] [Q&A]
Dear all,
what is the fast way to implement authentication and authorization (giving write access only to specific users) in stac-fastapi? Any suggestion?

Thank you very much.

**Comment by jamesfisher-geo** [comment]
Basic auth has been implemented in [stac-fastapi-elasticsearch-opensearch](https://github.com/stac-utils/stac-fastapi-elasticsearch-opensearch?tab=readme-ov-file#basic-auth) and in [stac-fastapi-mongo](https://github.com/Healy-Hyperspatial/stac-fastapi-mongo?tab=readme-ov-file#basic-auth)

**Comment by jonhealy1** [comment]
We have a draft pr too in stac-fastapi-elasticsearch-opensearch for OAuth 2

**Comment by g2giovanni** [comment]
Unfortunately, at the moment, we're using [stac-fastapi-pgstac](https://github.com/stac-utils/stac-fastapi-pgstac) as the backend.

**Reply by vincentsarago** [reply]
have you looked at https://fastapi.tiangolo.com/tutorial/security/ and https://fastapi.tiangolo.com/advanced/security/ 

it's pretty straightforward to set Authentification on fastapi application, using dependency injection or middleware.

in stac-fastapi we have https://github.com/stac-utils/stac-fastapi/blob/main/stac_fastapi/api/stac_fastapi/api/routes.py#L95-L127 which helps you add dependencies to specific route https://github.com/stac-utils/stac-fastapi/blob/1c3546a5b449edebc7587dab289f73224847c63e/stac_fastapi/api/tests/test_api.py#L90-L117
